src/PointwiseFunctions/GeneralRelativity/Psi4.cpp

```diff
@@ -28,11 +28,13 @@ void psi_4(const gsl::not_null<Scalar<ComplexDataVector>*> psi_4_result,
            const tnsr::ii<DataVector, 3, Frame>& spatial_metric,
            const tnsr::II<DataVector, 3, Frame>& inverse_spatial_metric,
            const tnsr::I<DataVector, 3, Frame>& inertial_coords) {
-  Variables<tmpl::list<::Tags::TempScalar<0>, ::Tags::TempI<0, 3, Frame>,
-                       ::Tags::TempI<1, 3, Frame>, ::Tags::Tempi<0, 3, Frame>,
-                       ::Tags::Tempij<0, 3, Frame>, ::Tags::Tempii<0, 3, Frame>,
-                       ::Tags::Tempii<1, 3, Frame>, ::Tags::TempIj<0, 3, Frame>,
-                       ::Tags::TempII<0, 3, Frame>>>
+  Variables<
+      tmpl::list<::Tags::TempScalar<0>, ::Tags::TempScalar<1>,
```

Comment:
Just so I understand, can you please explain to me why you need the additional temporaries?

Reply:
I think the additional TempScalar can be safely removed, however the additional TempI's are needed since TempI<0 = r_hat, TempI<1 = x_coord then changed to y_coord when not needed anymore, TempI<2 = x_hat and finally TempI<3 = y_hat_not_complex. I carry out the computation for y_hat in non-complex DataVector form for easier computation and then switch it to a ComplexDataVector later on.